### PR TITLE
305 Migrate to TypeScript

### LINKED ISSUE #305: Migrate to TypeScript

### PR DESCRIPTION
**Pull request by cdierkens**
Opening this PR for visibility. Still quite a bit of files to work through.

Closes #305

### CODE REVIEW

File: web/src/icons/index.tsx [hunk 1]
```diff
@@ -0,0 +1,142 @@
...
+import Shield from "./shield.svg";
+import ShilowCabinetry from "./shilow-cabinetry.svg";
+
+type Icon = React.FC<{
+  className?: string;
+}>;
+
+const baseClass = tw`p-1 fill-current stroke-current inline-block`;
+const iconBaseClass = tw`fill-none stroke-3`;
+const roundedBaseClass = tw`rounded-full`;
```
Comment (cdierkens) [inline web/src/icons/index.tsx:34]: This plus `"tailwindCSS.experimental.classRegex": ["tw`([^`]*)"]` in the workspace settings adds IntelliSense for places we want store class names in a variable.

File: web/src/icons/index.tsx [hunk 2]
```diff
@@ -0,0 +1,142 @@
...
+
+// Bluebird
+export const BirdIcon: Icon = ({ className }) => (
+  <Bird className={clsx(baseClass, className)} />
```
Comment (cdierkens) [inline web/src/icons/index.tsx:38]: `clsx` lets us add class names together and throw out falsy values like `undefined`

File: web/src/lib/index.ts [hunk 3]
```diff
@@ -1,2 +1 @@
-export * from "./helpers";
```
Comment (cdierkens) [inline web/src/lib/index.ts:1]: nothing imported from this file as far as I could tell

File: web/src/icons/index.tsx (hunk 1 above)
Comment (sonofab1rd) [reply web/src/icons/index.tsx:34]: Oooh, I might want to add that to The Basement of Stinkology too!

File: .github/workflows/pull_request_closed.yml [hunk 4]
```diff
@@ -11,6 +11,21 @@ env:
...
+    steps:
+      - uses: actions/checkout@v2
+      - uses: actions/setup-node@v1
+        with:
+          node-version: "14"
+
+      - name: Deploy Studio Graphql Server
+        run: |
+          npm ci
+          npx sanity graphql deploy --no-playground --dataset development
```
Comment (sonofab1rd) [inline .github/workflows/pull_request_closed.yml:26]: Should that be the production data set?

**Comment by cdierkens** [comment]
The secret sauce to get the vscode plug-in to do syntax highlighting is in
‘.vscode/settings.json’

**Review by sonofab1rd — COMMENTED** [review]
@cdierkens Just wondering about the Deploy Studio Graphql Server step in pulll_request_closed.yml!

File: .github/workflows/pull_request_closed.yml (hunk 4 above)
Comment (cdierkens) [reply .github/workflows/pull_request_closed.yml:26]: Before deploy previews the development set graphql API was deployed in the pull request action. 

This is a way to keep the development api in sync with production. This way you can keep your branches up to date with the main branch and still develop against the development API. There are also graphql API deploys based on the tags so you can target that when working on a PR.

I'm on mobile bet I'll give examples of that later

Comment (cdierkens) [reply .github/workflows/pull_request_closed.yml:26]: For instance this pr is deploy in the [`pull_request` action](https://github.com/cdierkens/bluebird-cabinetry-and-design/runs/2318607902?check_suite_focus=true):
```
Dataset: development
Tag: pull-request-809

GraphQL API deployed to:
https://5sj6awst.api.sanity.io/v1/graphql/development/pull-request-809
```

Here's a diagram of how I think about what sanity graphql APIs get deployed
bluebird-sanity-graphql-deploys

**Pull request merged**